### PR TITLE
feat: hpcai opensora 1.2 - VAE 3D training

### PR DESCRIPTION
**Pull request by SamitHuang**
# What does this PR do?

Fixes # (issue)

Adds # (feature)
VAE 3D training for hpcai opensora 1.2 including:
- 3 stage training with different loss config
- mixed video and image training

Evaluation are done on UCF-101 dataset, resulting in PSNR 29

TODOs:
- [ ] train with random number of frames. Dynamic shape support will be done in next PR

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/opensora_hpcai/opensora/models/vae/vae.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def encode_with_moments_output(self, x):
         """For latent caching usage"""
         h = self.encoder(x)
         moments = self.quant_conv(h)
-        mean, logvar = self.split(moments, moments.shape[1] // 2, 1)
+        mean, logvar = mint.split(moments, moments.shape[1] // 2, 1)
```
Comment (CaitinZhao) [inline examples/opensora_hpcai/opensora/models/vae/vae.py:54]: Why must be mint?

File: examples/opensora_hpcai/tools/convert_vae1.2.py
Comment (hadipash) [inline examples/opensora_hpcai/tools/convert_vae1.2.py:1]: what's the difference with `tools/convert_vae_3d.py`?

File: examples/opensora_hpcai/configs/vae/train/stage3.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+use_recompute: True
+
+epochs: 400
+ckpt_save_interval: 100
+init_loss_scale: 1.
+
+scheduler: "constant"
+use_ema: False
+
+output_path: "outputs/vae_stage2"
```
Comment (CaitinZhao) [inline examples/opensora_hpcai/configs/vae/train/stage3.yaml:46]:
```suggestion
output_path: "outputs/vae_stage3"
```

File: examples/opensora_hpcai/requirements.txt [hunk 3]
```diff
@@ -18,3 +18,4 @@ tokenizers
 sentencepiece
 transformers
 pyav
+mindcv
```
Comment (zhanghuiyao) [inline examples/opensora_hpcai/requirements.txt:21]: 是否要指定特定版本

File: examples/opensora_hpcai/run_train_vae_dynamic.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
+export ASCEND_RT_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
+
+# dynamic shape acceleration
+export MS_DEV_ENABLE_KERNEL_PACKET=on
```
Comment (zhanghuiyao) [inline examples/opensora_hpcai/run_train_vae_dynamic.sh:4]: 这个参数的做用是啥 是否动态shape下都需要打开

File: examples/opensora_hpcai/run_vae_recons.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    --num_frames 33 \
+    --dtype bf16 \
+    --video_folder ../videocomposer/datasets/webvid5 \
+
+	# --ckpt_path models/OpenSora-VAE-v1.2/model.ckpt \
+    # --ckpt_path outputs/vae_stage2.ckpt \
+	# --device_target GPU \
+    # --crop_size 256 \
+	# --ckpt_path /home/mindocr/yx/mindone/examples/opensora_hpcai/models/v1.2/vae.ckpt \
+	# --ckpt_path /home/mindocr/yx/mindone/examples/opensora_hpcai/models/sd-vae-ft-ema.ckpt \
```
Comment (zhanghuiyao) [inline examples/opensora_hpcai/run_vae_recons.sh:14]: 缩紧建议可以规整下

**Review by zhanghuiyao — APPROVED** [review]

File: examples/opensora_hpcai/README.md [hunk 6]
```diff
@@ -759,7 +761,74 @@ Here are some generation results after fine-tuning STDiT on a subset of WebVid d
...
+You can change the `csv_path` and `video_folder` to evaluate on your own data.
+
+Here, we report the training performance and evaluation results on the UCF-101 dataset.
+
+| Model       | Context      | jit_level | Precision | BS | NPUs | Resolution(framesxHxW) | Train T. (s/step) |    PSNR   |   SSIM  |
+|:------------|:-------------|:--------|:---------:|:--:|:----:|:----------------------:|:-----------------:|:-----------------:|:-----------------:|
+| STDiT2-XL/2 | D910\*-[CANN C18(0705)](https://repo.mindspore.cn/ascend/ascend910/20240705/)-[MS2.3](https://www.mindspore.cn/install) |    O1  |    BF16   |  1 |  8   |       17x256x256      |       0.97        |    29.29      |    0.88    |
+> Context: {G:GPU, D:Ascend}{chip type}-{mindspore version}.
+
+Note that we train with mixed video ang image strategy i.e. `--mixed_strategy=mixed_video_image` for stage 3 instead of random number of frames (`mixed_video_random`). Random frame training will be supported in the future.
```
[the remark this answers is not in the document]
Comment (SamitHuang) [reply examples/opensora_hpcai/README.md:830]: 确实，目前是对齐torch的参数名

File: examples/opensora_hpcai/README.md [hunk 7]
```diff
@@ -759,7 +761,74 @@ Here are some generation results after fine-tuning STDiT on a subset of WebVid d
...
+    ```
+    python tools/convert_vae1.2.py --src /path/OpenSora-VAE-v1.2/models.safetensors --target models/OpenSora-VAE-v1.2/sdxl_vae.ckpt
+    ```
+
+- Download lpips mindspore checkpoint from [here](https://download-mindspore.osinfra.cn/toolkits/mindone/autoencoders/lpips_vgg-426bf45c.ckpt) and put it under 'models/'
+
+
+### Data Preprocess
+If you want to train your own VAE, we need to prepare data in the csv following the [data processing](#data-processing) pipeline, then run the following commands.
+Note that you need to adjust the number of trained epochs (`epochs`) in the config file accordingly with respect to your own csv data size.
```
[the remark this answers is not in the document]
Comment (SamitHuang) [reply examples/opensora_hpcai/README.md]: fixed

File: examples/opensora_hpcai/configs/vae/train/stage3.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+use_real_rec_loss: True
+use_z_rec_loss: False
+use_image_identity_loss: False
+mixed_strategy: "mixed_video_image"  # TODO: use mixed_video_random after dynamic shape adaptation
+mixed_image_ratio: 0.2
+
+# data
+dataset_name: "video"
+csv_path: "../videocomposer/datasets/webvid5_copy.csv"
+video_folder: "../videocomposer/datasets/webvid5"
```
[the remark this answers is not in the document]
Comment (SamitHuang) [reply examples/opensora_hpcai/configs/vae/train/stage3.yaml:18]: 视频文件比较大，避免增大repo

File: examples/opensora_hpcai/requirements.txt (hunk 3 above)
Comment (SamitHuang) [reply examples/opensora_hpcai/requirements.txt]: fixed

File: examples/opensora_hpcai/run_train_vae_dynamic.sh (hunk 4 above)
Comment (SamitHuang) [reply examples/opensora_hpcai/run_train_vae_dynamic.sh]: yes

File: examples/opensora_hpcai/run_vae_recons.sh (hunk 5 above)
Comment (SamitHuang) [reply examples/opensora_hpcai/run_vae_recons.sh]: rm

File: examples/opensora_hpcai/opensora/models/vae/vae.py (hunk 1 above)
Comment (SamitHuang) [reply examples/opensora_hpcai/opensora/models/vae/vae.py]: fixed

File: examples/opensora_hpcai/tools/convert_vae1.2.py
Comment (SamitHuang) [reply examples/opensora_hpcai/tools/convert_vae1.2.py:1]: rm redundant file

**Review by vigo999 — APPROVED** [review]

**Pull request merged**